fapolicy_analyzer/ui/faprofiler.py

```diff
@@ -56,51 +110,69 @@ def __init__(self, msg="Unknown error",
 
 
 class FaProfSession:
-    def __init__(self, dictProfTgt, faprofiler=None):
+    def __init__(self, dictProfTgt, instance=0, faprofiler=None):
         logging.debug(f"faProfSession::__init__({dictProfTgt}, {faprofiler})")
         self.throwOnInvalidSessionArgs(dictProfTgt)
-
-        # Executable command and arguments
-        self.execPath = dictProfTgt["executeText"]
-        self.execArgs = dictProfTgt["argText"]
-        self.listCmdLine = [self.execPath] + self.execArgs.split()
+        self.timeStamp = None
 
         # euid, working directory and environment variables
         self.user = dictProfTgt["userText"]
-        self.pwd = dictProfTgt["dirText"]
+
+        # Set pwd - Use current dir for pwd if not supplied by user
+        tmp_d = dictProfTgt["dirText"]
+        self.pwd = tmp_d if tmp_d else os.getcwd()
```

Comment:
If you are worried about this being empty this would be the safest and simplest way to do that. In fact all your initial accesses to the `dictProfTgt` dictionary should use the `.get()` method in case a bad dictionary is passed in that doesn't have all the keys needed.

```suggestion
        self.pwd = dictProfTgt.get("dirText") or os.getcwd()
```

Reply:
I do this multi-line thing so that my single lines are fully visible on an 80 char screen i.e. avoiding

> `<tab><tab>...<tab>self.pwd = dictProfTgt["dirText"] if dictProfTgt["dirText"]  else os.getcwd()`

which overflows. Was not worried about missing K=V pairs; if it's missing I want it to crash. If any of the UI fields are empty, the assoiated dict value is an empty string.

In any event, your suggestion: `self.pwd = dictProfTgt.get("dirText") or os.getcwd()` is better. Changed it.